codec-http2/src/main/java/io/netty/handler/codec/http2/InboundHttpToHttp2Adapter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http2;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.codec.http.FullHttpMessage;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpScheme;
+
+/**
+ * Translates HTTP/1.x object reads into HTTP/2 frames.
+ */
+public class InboundHttpToHttp2Adapter extends ChannelInboundHandlerAdapter {
+    private final Http2Connection connection;
+    private final Http2FrameListener listener;
+
+    public InboundHttpToHttp2Adapter(Http2Connection connection, Http2FrameListener listener) {
+        this.connection = connection;
+        this.listener = listener;
+    }
+
+    private int getStreamId(HttpHeaders httpHeaders) {
+        return httpHeaders.getInt(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text(),
+            connection.remote().nextStreamId());
+    }
+
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+        if (msg instanceof FullHttpMessage) {
```

Comment:
We could support types other than `FullHttpMessage`, but I don't need it and the class was a distraction to begin with. It can easily be enhanced in the future and I wrote it only to support HTTP 1 upgrade, which will always have a `FullHttpMessage`.

Eventually, I would expect to stop using the class in favor of a new class that maps to the new HTTP/2 frames.